packages/nocodb/src/lib/cache/RedisCacheMgr.ts

```diff
@@ -125,9 +131,22 @@ export default class RedisCacheMgr extends CacheMgr {
     // e.g. arr = ["nc:<orgs>:<scope>:<model_id_1>", "nc:<orgs>:<scope>:<model_id_2>"]
     const arr = (await this.get(key, CacheGetType.TYPE_ARRAY)) || [];
     log(`RedisCacheMgr::getList: getting list with key ${key}`);
-    return Promise.all(
-      arr.map(async (k) => await this.get(k, CacheGetType.TYPE_OBJECT))
-    );
+
+    const isNoneList = arr.length && arr[0] === 'NONE';
+
+    if (isNoneList) {
+      return Promise.resolve({
+        list: [],
+        isNoneList,
+      });
+    }
+
+    return {
+      list: await Promise.all(
+        arr.map(async (k) => await this.get(k, CacheGetType.TYPE_OBJECT))
+      ),
+      isNoneList,
+    };
```

Comment:
Instead of relying on the first value in the array `'NONE'`, what about checking the subkeys array, If it's defined it will return an array otherwise it will be undefined/null. Can we use the following method?

```
 const arr = (await this.get(key, CacheGetType.TYPE_ARRAY));
    log(`RedisCacheMgr::getList: getting list with key ${key}`);

    const isNoneList = !!arr;
```

The current solution is fine and the only drawback is we can't save `NONE` as the first value in the array which is never going to happen in our current scenario.

Reply:
> the only drawback is we can't save NONE as the first value in the array which is never going to happen in our current scenario.

It's in `setList`. We save `NONE` when the input list is empty. 

Will test your approach.

Reply:
Just tested a bit. In this way, we never trigger metaList2. Let's keep it as it is first.